FreeRTOS/FreeRTOS-Plus-TCP
# Add libslirp NetworkInterface implementation

Add libslirp NetworkInterface implementation

Description
-----------
Libslirp is used by qemu to provide userspace NAT networking. This serves as a replacement for the current libpcap / winpcap NetworkInterface but does not require elevated privileges or selecting a particular interface to bind to.

In this implementation, incoming connections are not supported. This can be extended in the future by calling the libslirp port forward api calls during socket bind() and close() operations.

Test Steps
-----------
Tested with libslirp git revision 38638382c8f3dc2523468f0c54014f4fdf0aa38f on MacOS / AppleClang and Windows/MSVC (with glib from vcpkg).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
